frontend/src/api/users.ts

```diff
@@ -0,0 +1,10 @@
+import axios from "axios";
+import { SignUpData } from "../common/api.types";
+import { SignUpResponse, User } from "../common/internal";
+
+const baseUrl = "api/users/sign-up";
+
+export const create = async (data: SignUpData): Promise<User> => {
+  const response: SignUpResponse = await axios.post(baseUrl, data);
```

Comment:
Should put `SignUpResponse` in to the `post` generic type:
```suggestion
  const response = await axios.post<SignUpResponse>(baseUrl, data);
```